src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -760,52 +783,36 @@ private void CommitCore(IUIThreadOperationContext operationContext, bool preview
                     }
                 }
 
-                // The user hasn't cancelled by now, so we're done waiting for them. Off to
-                // rename!
+                // The user hasn't canceled by now, so we're done waiting for them. Off to rename!
                 using var _ = operationContext.AddScope(allowCancellation: false, EditorFeaturesResources.Updating_files);
 
-                Dismiss(rollbackTemporaryEdits: true);
-                CancelAllOpenDocumentTrackingTasks();
-
-                _triggerView.Caret.PositionChanged += LogPositionChanged;
-
-                ApplyRename(newSolution, operationContext);
-
-                LogRenameSession(RenameLogMessage.UserActionOutcome.Committed, previewChanges);
-
-                EndRenameSession();
-
-                _triggerView.Caret.PositionChanged -= LogPositionChanged;
-
-                void LogPositionChanged(object sender, CaretPositionChangedEventArgs e)
-                {
-                    try
+                DismissUIAndRollbackEditsAndEndRenameSession(
+                    RenameLogMessage.UserActionOutcome.Committed, previewChanges,
+                    () =>
                     {
-                        throw new InvalidOperationException("Caret position changed during application of rename");
-                    }
-                    catch (InvalidOperationException ex) when (FatalError.ReportAndCatch(ex))
-                    {
-                        // Unreachable code due to ReportAndCatch
-                        Contract.ThrowIfTrue(true);
-                    }
-                }
+                        var error = TryApplyRename(newSolution);
+
+                        if (error is not null)
+                        {
+                            operationContext.TakeOwnership();
+                            var notificationService = _workspace.Services.GetService<INotificationService>();
+                            notificationService.SendNotification(
+                                error, EditorFeaturesResources.Rename_Symbol, NotificationSeverity.Error);
+                        }
```

Comment:
single place in inline-rename, and single place in rename-code-action where we notify the user of an error we couldnt' handle.